specs-go/chain/chainid.go

```diff
@@ -0,0 +1,46 @@
+package chain
+
+import "github.com/docker/distribution/digest"
+
+// ChainID takes a slice of digests and returns the ChainID corresponding to
+// the last entry. Typically, these are a list of layer DiffIDs, with the
+// result providing the ChainID identifying the result of sequential
+// application of the preceeding layers.
+func ChainID(dgsts []digest.Digest) digest.Digest {
+	chainIDs := make([]digest.Digest, len(dgsts))
+	copy(chainIDs, dgsts)
+	ChainIDs(chainIDs)
+
+	if len(chainIDs) == 0 {
+		return ""
```

Comment:
Neither the [current definition][1] nor any of the alternative phrasings proposed in #482 define chainIDs for empty arrays.  I think we want:

    func ChainID(digests []digest.Digest) (digest.Digest, error)

with an error being returned when `len(digests) == 0`.

[1]: https://github.com/opencontainers/image-spec/blob/656fb2f82121e8bd611b5aae4a6a3471e1bfe078/config.md#layer-chainid